### PR TITLE
Feature/execute custom action

### PR DESCRIPTION
**Pull request by B3zo0**
In this pr:
- fixed a bug in PathHelper.getSegments method (/root/content/some(1).pdf gave back wrong segment)
- fixed a console error on app start
- created custom actions for crud (Create, Load, LoadCollection, Update, **Remove** - _not Delete_)
  - create can be executed with 2 parameters: 
    - content -> object that properties are the fields of the content eg.: `{ "Name": "someContent"}`
    - contentType -> string, eg.: "Task", "File", "Folder"
  - update can be executed with 1 parameter: content
    - content -> object that properties are the fields of the content eg.: `{ "Name": "someContent"}`
  - remove can be executed with 1 parameter
    - isPermanent -> boolean defaults to false if it is not specified
-  fixed a bug that closed the execute action window when an error came back from the server

### CODE REVIEW

File: apps/sensenet/src/components/dialogs/execute-action.tsx [hunk 1]
```diff
@@ -63,6 +64,76 @@ export const ExecuteActionDialog: React.FunctionComponent = () => {
...
+          return await repo.post({
+            contentType: parsedBody.contentType,
+            parentPath: actionValue.content.IsFolder
+              ? actionValue.content.Path
+              : PathHelper.getParentPath(actionValue.content.Path),
+            content: parsedBody.content,
+          })
+        }
+        case 'Remove':
+          return await repo.delete({ idOrPath: actionValue.content.Id, permanent: false })
```
Comment (herflis) [inline apps/sensenet/src/components/dialogs/execute-action.tsx]: Make ```permanent``` adjustable

Comment (B3zo0) [reply apps/sensenet/src/components/dialogs/execute-action.tsx]: ✅

File: apps/sensenet/src/components/dialogs/execute-action.tsx [hunk 2]
```diff
@@ -63,14 +67,89 @@ export const ExecuteActionDialog: React.FunctionComponent = () => {
...
+          return await repo.post({
+            contentType: parsedBody.contentType,
+            parentPath: actionValue.content.IsFolder
+              ? actionValue.content.Path
+              : PathHelper.getParentPath(actionValue.content.Path),
+            content: parsedBody.content,
+          })
+        }
+        case 'Remove': {
+          const { isPermanent } = JSON.parse(postBody)
```
Comment (herflis) [inline apps/sensenet/src/components/dialogs/execute-action.tsx]:
```suggestion
          const { permanent} = JSON.parse(postBody)
```

File: apps/sensenet/src/components/dialogs/execute-action.tsx [hunk 3]
```diff
@@ -63,14 +67,89 @@ export const ExecuteActionDialog: React.FunctionComponent = () => {
...
+          return await repo.delete({
+            idOrPath: actionValue.content.Id,
+            permanent: isPermanent == null ? false : isPermanent,
```
Comment (herflis) [inline apps/sensenet/src/components/dialogs/execute-action.tsx]:
```suggestion
            permanent: permanent== null ? false : permanent,
```

File: apps/sensenet/src/services/CommandProviders/CustomActionCommandProvider.ts [hunk 4]
```diff
@@ -78,4 +87,22 @@ export class CustomActionCommandProvider implements CommandProvider {
...
+        return {
+          parameters: [
+            { name: 'contentType', type: 'string', required: true },
+            { name: 'content', type: 'object', required: true },
+          ],
+        }
+      case 'Update':
+        return { parameters: [{ name: 'content', type: 'object', required: true }] }
+      case 'Remove':
+        return { parameters: [{ name: 'isPermanent', type: 'boolean', required: true }] }
```
Comment (B3zo0) [inline apps/sensenet/src/services/CommandProviders/CustomActionCommandProvider.ts]:
```suggestion
        return { parameters: [{ name: 'permanent', type: 'boolean', required: false }] }
```

**Pull request merged**